tests/test.rs

```diff
@@ -691,6 +720,35 @@ fn test_integration_guess_single_file() {
 
     check_equal_covdir(
         &read_file(&json_path),
-        &run_grcov(vec![&zip_path], &PathBuf::from(""), "covdir"),
+        &run_grcov(vec![&zip_path], &PathBuf::from(""), "covdir", Vec::new()),
     );
 }
+
+#[test]
+fn test_integration_service_job_number() {
```

Comment:
We could add a simpler test for service_job_number, rather than an integration test. WDYT?
An integration test would make sense if you had to use git, but in this case it's pretty simple.

Reply:
Making it a normal test would also solve the issue you were getting with the different compiler versions

Reply:
Oh, this makes perfect sense! I guess I got hung up on integration tests because of #370: I kinda viewed testing here as a practice for that issue. I'll update this PR tomorrow.

Reply:
Out of time today, sorry. Will do it on the weekend for sure!

Reply:
Done. I actually ran into another problem with a new test: Coveralls generates MD5 digests for source files, but the test doesn't create the files on disk, so grcov generates UUIDs instead. These change from run to run, failing the tests.

I thought of two fixes: 1) create the files on disk; 2) don't try to compare whole grcov output to some "golden standard; instead, check just the `service_job_id` key. I implemented the latter because it's more in line with this being a unit test.

However, I'm now doubtful about the whole approach. Integration test worked end-to-end, ensuring that both `--service-job-number` and `--service-job-id` exist and do the thing we expect. The new test is less useful: we could break the CLI options, but the test will still succeed.

I feel like rolling back to the integration test, and implementing the narrow check there (just `service_job_id` key) would be an ideal solution. What do you think, @marco-c?

Reply:
@marco-c, ping.

Reply:
Sorry for the delay! I think it's OK if we don't test the CLI options, as they are simple and we are going to remove --service-job-number soon anyway, so don't worry about it ;)